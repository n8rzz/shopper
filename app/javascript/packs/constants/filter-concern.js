export const FILTER_CONCERN = {
    ALPHA: 'ALPHA',
    DEPARTMENT: 'DEPARTMENT',
    FILTER: 'FILTER',
    ITEM: 'ITEM',
    MEAL: 'MEAL',
};

export const FILTER_LABEL = {
    [FILTER_CONCERN.ALPHA]: 'Alpha',
    [FILTER_CONCERN.DEPARTMENT]: 'Department',
    [FILTER_CONCERN.FILTER]: 'Filter',
    [FILTER_CONCERN.ITEM]: 'Item',
    [FILTER_CONCERN.MEAL]: 'Meal',
};
