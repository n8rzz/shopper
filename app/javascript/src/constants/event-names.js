export const EVENT_NAME = {
    NOTICE_SUCCESS: 'NOTICE_SUCCESS',
    NOTICE_WARNING: 'NOTICE_WARNING',
    NOTICE_ERROR: 'NOTICE_ERROR',
};
