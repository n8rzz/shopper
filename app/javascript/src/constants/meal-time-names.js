export const MEAL_TIME_NAME = {
    MORNING: 'morning',
    AFTERNOON: 'afternoon',
    EVENING: 'evening',
    NONE: 'none',
};
