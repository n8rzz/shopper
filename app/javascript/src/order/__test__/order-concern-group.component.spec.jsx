import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import { OrderConcernGroup } from '../order-concern-group.component';
// TODO: implement tests for all states of the filter
import { FILTER_CONCERN } from '../../constants/filter-concern';
import {
    assemblyListMock,
    assemblyMapMock,
    csrfMock,
    departmentListMock,
    departmentMapMock,
    locationDepartmentListMock,
    orderItemListMock,
    itemListMock,
    itemMapMock,
} from '../../__mocks__/mocks';

describe('OrderConcernGroup', () => {
    test('renders correctly with valid props', () => {
        const component = shallow(<OrderConcernGroup
            assemblies={assemblyListMock}
            assemblyMap={assemblyMapMock}
            concern={FILTER_CONCERN.ITEM}
            csrf={csrfMock}
            departmentMap={departmentMapMock}
            departments={departmentListMock}
            itemMap={itemMapMock}
            items={itemListMock}
            locationDepartments={locationDepartmentListMock}
            orderItems={orderItemListMock}
            onClickIsPickedHandler={sinon.spy()}
            onClickRemoveItemHandler={sinon.spy()}
        />);

        expect(component).toMatchSnapshot();
    });
});
