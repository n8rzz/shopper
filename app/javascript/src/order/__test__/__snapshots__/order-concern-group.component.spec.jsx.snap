// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderConcernGroup renders correctly with valid props 1`] = `""`;
